cloud-controller-manager/do/firewall_controller_test.go

```diff
@@ -491,7 +491,7 @@ func TestFirewallController_NoDataRace(t *testing.T) {
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		fc.Run(context.TODO(), ctx.Done(), time.Duration(0))
+		fc.Run(1, ctx.Done(), time.Duration(0))
```

Comment:
Currently I don't see us using more than one thread/goroutine for handling Service events in the context of the worker firewall: by design, we always look at all (NodePort) Services on each run.

Overall, I'd remove the `threadiness` variable and instantiate a single worker goroutine in the controller only.

Reply:
> instantiate a single worker goroutine in the controller only

Are you suggesting to take out the threadiness variable and the loop that references it, and leave the `go wait.Until(func)...`.

_or_

Add a wrapping `go func()` so that this is included? @timoreimann